### DISCUSSION TITLE
[FR] Add Conditional Feature based on Device

### DISCUSSION THREAD

**Discussion by LanceZeligman-KR** [NemesisRE/kiosk-mode#280] [Ideas]
**Is your feature request related to a problem? Please describe.**

Being able to conditionally enable or disable kiosk mode based on what device is accessing a dashboard would be awesome.

**Describe the solution you'd like**

I have a dashboard that I like to use both for my nest display and my other various devices. Being able to selectively enable kiosk mode based on IP address or some other device identifier would be an extremely useful feature for me.

**Describe alternatives you've considered**

I'm currently playing around with the existing conditionals and the query urls like ?kiosk but looks like that didn't do what I was hoping it would.

**Additional context**
N/A

**Comment by elchininet** [comment]
Hi @LanceZeligman-KR,
Please, read the description in [this pull request](https://github.com/NemesisRE/kiosk-mode/pull/274) so you can get what is the intention behind it. With those changes you will be able to use client side variables like [navigator.userAgentData](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorUAData/platform) to set some `kiosk-mode` options dynamically.

`navigator.userAgentData` will give you something like this:

<img width="542" alt="image" src="https://github.com/user-attachments/assets/6795e017-4e2d-4902-af2e-687532222724">

You can also get the user agent of the browser (more browser compatibility), which in my case is something like this:

```
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36
```

It is not merged yet because I am waiting for some folks that are testing it, but if you want to give it a try just use the gist that is at the begining of the description, in that way you can also help to test this new feature.

Regards

**Reply by LanceZeligman-KR** [reply]
That's awesome, thank you so much. I'll be glad to give it a try, if I have free time this weekend I'll let you know how it goes for me.

**Reply by lanceofwhichwedream** [reply]
Hello, this is lance again, I was accidentally on my work account. 

Is it possible to output the device array you're showing via the debug feature you're adding? My usecase is I am using the automatic casting plugin to send a dashboard to my nest hub, so I would need a way to find it's user agent.

**Reply by NemesisRE** [reply]
If you are casting, your casting source will contain those information

**Reply by elchininet** [reply]
@lanceofwhichwedream,

>Is it possible to output the device array you're showing via the debug feature you're adding? My usecase is I am using the automatic casting plugin to send a dashboard to my nest hub, so I would need a way to find it's user agent.

The debug feature is intended to display information in the browser console. If you want to cast a webpage you will not have access to the browser console I am affraid.

But to know the user agent of a device can be done with multiple webpages, for example:

https://www.whatsmyua.info/

Try to cast that webpage instead of your Home Assistant dashboard and check what user agent is displayed. 

If you don't want to change the settings in Home Assistant just use [url-cast-receiver](https://github.com/DeMille/url-cast-receiver):

https://demille.github.io/url-cast-receiver/

But knowing that you are casting a URL, what happens when you send to cast a URL using [the query strings](https://github.com/NemesisRE/kiosk-mode?tab=readme-ov-file#query-strings)?

@NemesisRE,

>If you are casting, your casting source will contain those information

I think that when casting a URL the casting device receives an order to open a webpage and it opens it using its own internal browser. I don't have any casting plugin installed in my Home Assistant to check which namespace is used, but if the `automatic casting plugin` uses the `urn:x-cast:com.url.cast` then it will send a message to the device to open a URL. I think that it is a matter of testing and checking what is displayed there.
